### PR TITLE
Add copy-container SEXP

### PR DESCRIPTION
**Pull request by jg18**
Per suggestion from @MjnMixael in #fred.

### CODE REVIEW

File: code/parse/sexp_container.cpp [hunk 1]
```diff
@@ -1311,6 +1320,68 @@ void sexp_clear_container(int node)
...
+	const char *dest_container_name = CTEXT(node);
+	auto *p_dest_container = get_sexp_container(src_container_name);
+
+	if (!p_dest_container) {
+		report_nonexistent_container("Copy-container", dest_container_name);
+		return;
+	}
+	auto &dest_container = *p_dest_container;
+
+	if (src_container.get_non_persistent_type() != dest_container.get_non_persistent_type()) {
```
Comment (JohnAFernandez) [inline code/parse/sexp_container.cpp:1348]: Seems pretty straight forward.  The only thing is the name of this function `get_non_persistent_type`.  That just means what type of container type it is?

Comment (jg18) [reply code/parse/sexp_container.cpp:1348]: Yeah, get a container's type (function return type is `enum class ContainerType`) with persistence flags zeroed out.

See the SEXP help text and also the use case described in #fred for why this is done.

Comment (jg18) [reply code/parse/sexp_container.cpp:1348]: Admittedly `get_non_persistent_Type()` wasn't the best function name. 🤷‍♂️

Comment (JohnAFernandez) [reply code/parse/sexp_container.cpp:1348]: Go figure

**Review by JohnAFernandez — APPROVED** [review]
To my "sexp-amateur" eyes, it looks fine.

**Comment by jg18** [comment]
The latest version passes manual tests by @MjnMixael.

**Review by JohnAFernandez — APPROVED** [review]

**Pull request merged**